spec/workers/postcodes_collection_worker_spec.rb

```diff
@@ -0,0 +1,24 @@
+RSpec.describe PostcodesCollectionWorker do
+  describe "#perform" do
+    def perform
+      described_class.new.perform
+    end
+
+    context "where there are postcodes to be refreshed" do
+      before do
+        allow(Postcode).to receive(:select).and_return(%w[E18QS E18QL])
+      end
+
+      it "creates a worker for each postcode" do
+        expect(ProcessPostcodeWorker).to receive(:perform)
+          .with("E18QS")
+        expect(ProcessPostcodeWorker).to receive(:perform)
+          .with("E18QL")
+
+        expect(subject).to receive(:sleep).exactly(2).times
+
+        subject.perform(test: true)
```

Comment:
I'd suggest inverting the parameter here - change it to `run_continuously: false` (but set to `true` by default).
We may legitimately want to only do single run-throughs someday, so this would give us that feature.
Rather than telling the application code "this is a test, so do something different", we're specifying in the test how we want the application code to run.
Hope that makes sense!